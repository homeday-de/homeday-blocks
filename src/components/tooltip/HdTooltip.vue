<template>
  <span ref="tooltip" class='hd-tooltip'></span>
</template>

<script>
export default {
  name: 'hd-tooltip',
  mounted() {
    window.tooltipElement = this.$refs.tooltip;
  },
};
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped lang="scss">
@import 'hd-blocks/styles/mixins.scss';

.hd-tooltip {
  position: absolute;
  top: 0;
  min-width: 80px;
  text-align: center;
  font-size: 12px;
  line-height: 12px;
  padding: 4px 8px;
  background-color: $nevada;
  border-radius: 5px;
  color: white;
  pointer-events: none;
  opacity: 0;
  z-index: 1000;
  box-shadow: 0 2px 5px 0 rgba(0,0,0,.2);
  &.show {
    animation: fadeInBottom .4s ease-in-out forwards;
    &.left {
      animation: fadeInLeft .4s ease-in-out forwards;
    }
  }
  @keyframes fadeInBottom {
    from { opacity: 0; transform: translateY(-4px); }
    to { opacity: 1; transform: translateY(0px); }
  }
  @keyframes fadeInLeft {
    from { opacity: 0; transform: translateX(4px); }
    to { opacity: 1; transform: translateX(0px); }
  }
}
</style>
