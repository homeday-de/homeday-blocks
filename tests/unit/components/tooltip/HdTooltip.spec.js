import { mount } from '@vue/test-utils';
import HdTooltip from '@/components/tooltip/HdTooltip.vue';

describe('HdTooltip', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(HdTooltip);
  });

  test('renders component', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('renders a span', () => {
    const span = wrapper.find('span');

    expect(span.exists()).toBe(true);
  });
});
