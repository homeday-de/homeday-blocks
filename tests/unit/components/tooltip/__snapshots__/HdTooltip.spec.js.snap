// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HdTooltip the component is rendered 1`] = `<span class="hd-tooltip"></span>`;
